### DISCUSSION TITLE
RetroArch netplay issue with other OS

### DISCUSSION THREAD

**Discussion by phuctm97** [OnionUI/Onion#1225] [Q&A]
I was trying to see if it's possible to netplay using RetroArch in different OSes together.

I was testing using an Anbernic RG353VS running JELOS + Miyoo Mini Plus running Onion 4.2 + RetroArch on macOS.

TL;DR:

It works and it works consistently across many systems for the Anbernic RG353VS running JELOS + RetroArch on macOS. However, for the Miyoo Mini Plus running Onion 4.2, only NES works, for other systems, it consistently connects successfully then instantly disconnects.

In a word, it seems to be possible to netplay across OSes and the RetroArch Onion seems to be incompatible. Is it because Onion has different RetroArch version/build or differently configured? Would appreciate any pointers! Thank you ❤️

**Comment by schmurtzm** [comment]
Theorically to work Netplay requires the same Retroarch version, the same core version, the same rom and the same CPU architecture...
But indeed it some cases it works without respecting these parameters. TGB Dual (the GB core) should work correctly too with your mac.

What could help for more precise investigation is to report all the versions you use : for example 
SNES : **mac** RA version 1.1x , with core snes9x20xx version xxxx   / **Jelos** RA version 1.1x , with core snes9x20xx version xxxx  
With these info we could try to reproduce the behavior of jelos to improve the cross compatibility.

It would also help to complete [our compatibility file](https://docs.google.com/spreadsheets/d/1y1esG3HcNagPG8cy-ulZK_mhhUb93pqG5_XEtRRgKQ4/edit#gid=0) (we check the cross compatibility into it too).

**Comment by roo1234** [comment]
> I was trying to see if it's possible to netplay using RetroArch in different OSes together.
> 
> I was testing using an Anbernic RG353VS running JELOS + Miyoo Mini Plus running Onion 4.2 + RetroArch on macOS.
> 
> TL;DR:
> 
> It works and it works consistently across many systems for the Anbernic RG353VS running JELOS + RetroArch on macOS. However, for the Miyoo Mini Plus running Onion 4.2, only NES works, for other systems, it consistently connects successfully then instantly disconnects.
> 
> In a word, it seems to be possible to netplay across OSes and the RetroArch Onion seems to be incompatible. Is it because Onion has different RetroArch version/build or differently configured? Would appreciate any pointers! Thank you ❤️

I have the exact same problem. I've been testing with puyo puyo 2 megadrive (picodrive) and I can successfully connect between my trimui, R36S and xu10 with picodrive and same rom. With onion 4.3 on my miyoo+, Net play is connected successfully and just disconnects immediately, no extra info given. I've tried to disable the wifi option for games, start Net play in menu and RetroArch alone, hosting or joining a host, disable auto savestate, all the same. I will soon have a 2nd miyoo mini+ to try between them...

**Comment by roo1234** [comment]
Got a new mini+, the little devil only wants to net play with ones of its kind, it works and no instant disconnect. Not a fair player :(
